src/data/matrix/notation.lean

```diff
@@ -249,16 +249,16 @@ lemma vec3_add [has_add α] (a₀ a₁ a₂ b₀ b₁ b₂ : α) :
   ![a₀, a₁, a₂] + ![b₀, b₁, b₂] = ![a₀ + b₀, a₁ + b₁, a₂ + b₂] :=
 by rw [cons_add_cons, cons_add_cons, cons_add_cons, empty_add_empty]
 
-variable [semiring α]
-
-lemma smul_vec2 (x : α) (a₀ a₁ : α) :
+lemma smul_vec2 [has_mul α] (x : α) (a₀ a₁ : α) :
   x • ![a₀, a₁] = ![x • a₀, x • a₁] :=
 by rw [smul_cons, smul_cons, smul_empty]
 
-lemma smul_vec3 (x : α) (a₀ a₁ a₂ : α) :
+lemma smul_vec3 [has_mul α] (x : α) (a₀ a₁ a₂ : α) :
   x • ![a₀, a₁, a₂] = ![x • a₀, x • a₁, x • a₂] :=
 by rw [smul_cons, smul_cons, smul_cons, smul_empty]
 
+variable [semiring α]
```

Comment:
I think this actually only needs
```suggestion
variables [add_comm_monoid α] [has_mul α]
```

Reply:
I added it to the "experimental" commit.
https://github.com/leanprover-community/mathlib/pull/11429/commits/9c302d38e4ae40f038baa56db45dd585e00db2e6